python_modules/libraries/dagster-databricks/dagster_databricks/databricks_pyspark_step_launcher.py

```diff
@@ -269,6 +281,74 @@ def _get_step_records():
 
         return []
 
+    def _grant_permissions(self, log, databricks_run_id, request_retries=3):
+        api_client = self.databricks_runner.client.client.client
+
+        # Retrieve run info
+        cluster_id = None
+        for i in range(1, request_retries + 1):
+            run_info = self.databricks_runner.client.get_run(databricks_run_id)
+            # if a new job cluster is created, the cluster_instance key may not be immediately present in the run response
+            try:
+                cluster_id = run_info["cluster_instance"]["cluster_id"]
+                break
+            except:
+                log.warning(
+                    f"Failed to retrieve cluster info for databricks_run_id {databricks_run_id}. "
+                    f"Retrying {i} of {request_retries} times."
+                )
+                time.sleep(3)
+        if not cluster_id:
+            log.warning(
+                f"Failed to retrieve cluster info for databricks_run_id {databricks_run_id} "
+                f"{request_retries} times. Skipping permission updates..."
+            )
+            return
+
+        # Update job permissions
+        if "job_permissions" in self.permissions:
+            job_permissions = {"access_control_list": []}
+            for permission_level, accessors in self.permissions["job_permissions"].items():
+                for accessor in accessors:
+                    job_permissions["access_control_list"].append(
+                        {**accessor, **{"permission_level": permission_level}}
+                    )
+
+            job_id = run_info["job_id"]
+            endpoint = f"{api_client.url}2.0/permissions/jobs/{job_id}"
+            log.debug(f"Updating job permissions with following json: {job_permissions}")
+            response = requests.patch(
+                endpoint, headers=api_client.default_headers, data=json.dumps(job_permissions)
+            )
+            if not response.ok:
+                log.warning(
+                    f"Received invalid HTTP response: {response.status_code}.\n{response.text}"
+                )
+            else:
+                log.debug(f"Updated job permissions | Response: {response.text}")
+
+        # Update cluster permissions
+        if "new" in self.run_config["cluster"] and "cluster_permissions" in self.permissions:
+            cluster_permissions = {"access_control_list": []}
+            for permission_level, accessor in self.permissions["cluster_permissions"].items():
```

Comment:
this logic appears identical to the logic above, so this might be worth refactoring